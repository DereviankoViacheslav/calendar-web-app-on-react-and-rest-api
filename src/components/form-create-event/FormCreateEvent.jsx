import React from 'react';
import './FormCreateEvent.scss';
import PropTypes from 'prop-types';

function FormCreateEvent(props) {
  return (
    <form onSubmit={props.handleSubmit} className="popup event">
      <span onClick={props.onClosePopup} className="popup__btn-close">
        <i className="fas fa-times"></i>
      </span>
      <input name="name" required className="event__name" type="text" placeholder="Add title "
        value={props.name} onChange={props.onChange} />
      <div className="popup__color-picker">
        <i className="fas fa-palette"></i>
        <label className="popup__color-picker_label">
          <span className="popup__color-picker_label-text">Select a color</span>
          <input name="color" className="event__color-picker" type="color"
            value={props.color} onChange={props.onChange} />
        </label>
      </div>
      <div className="popup__picker">
        <span className="popup__picker-icon">
          <i className="far fa-clock"></i>
        </span>
        <input name="startDate" required className="event__date-start" type="date"
          value={props.startDate} onChange={props.onChange} />
        <input name="startTime" required className="event__time-start select" type="time" step="900"
          value={props.startTime} onChange={props.onChange} />
        <span className="decor-dash"></span>
        <input name="endTime" required className="event__time-end select" type="time" step="900"
          value={props.endTime} onChange={props.onChange} />
        <input name="endDate" required className="event__date-end" type="date"
          value={props.endDate} onChange={props.onChange} />
      </div>
      <div className="popup__description">
        <i className="fas fa-align-left"></i>
        <textarea name="description" className="event__description" placeholder="Add description"
          value={props.description} onChange={props.onChange} />
      </div>
      <div className="footer-popup">
        {props.isShowTrashIcon &&
          <span onClick={props.onDeleteEvent} className="event__btn-delete">
            <i className="fas fa-trash-alt"></i>
          </span>}
        <button className="event__btn-save">Save</button>
      </div>
    </form>
  );
}

FormCreateEvent.propTypes = {
  onChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onClosePopup: PropTypes.func.isRequired,
  onDeleteEvent: PropTypes.func.isRequired,
  startDate: PropTypes.string.isRequired,
  startTime: PropTypes.string.isRequired,
  endDate: PropTypes.string.isRequired,
  endTime: PropTypes.string.isRequired,
  color: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  isShowTrashIcon: PropTypes.bool.isRequired,
}

FormCreateEvent.defaultProps = {
  name: '',
  description: '',
}

export default FormCreateEvent;