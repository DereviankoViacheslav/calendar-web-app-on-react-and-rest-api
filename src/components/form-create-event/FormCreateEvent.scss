.popup {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 16px 16px 8px;
  border-radius: 8px;
  background-color: #fff;
  z-index: 9999;

  &__picker {
    display: flex;
    margin: 45px 0;
    align-items: center;
  }

  &__picker-icon {
    margin-left: 20px;
  }

  &__color-picker {
    margin: 45px 0 0 20px;
    text-align: left;
  }

  &__color-picker_label-text {
    margin-right: 15px;
    margin-left: 20px;
  }

  &__description {
    display: flex;
    margin-left: 20px;
  }

  &__btn-close {
    position: absolute;
    right: 20px;
    top: 20px;
    cursor: pointer;
  }
}

.event {

  &__name {
    border: none;
    border-bottom: 2px solid #d5d5d5;
    width: 79%;
    font-size: 20px;
    letter-spacing: 2px;
    outline: none;
    margin-top: 30px;
    margin-left: 55px;
  }

  &__color-picker {
    cursor: pointer;
  }

  &__date-start,
  &__date-end {
    padding: 2px;
    margin-left: 18px;
    border: none;
    background-size: contain;
    outline: none;
    width: 150px;
  }

  &__time-start,
  &__time-end {
    border: none;
    background-size: contain;
    outline: none;
    padding: 2px;
    background-color: #fff;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
  }

  &__description {
    height: 49px;
    width: 90%;
    margin-left: 20px;
    margin-bottom: 20px;
    border: none;
    outline: none;
    resize: none;
  }

  &__btn-save {
    position: absolute;
    right: 5px;
    height: 40px;
    padding-left: 20px;
    padding-right: 20px;
    background: royalblue;
    color: white;
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.22);
    border: none;
    border-radius: 5px;
    cursor: pointer;
  }

  &__btn-delete {
    height: 20px;
    margin-left: 20px;
    cursor: pointer;
  }
}

.footer-popup {
  position: relative;
  height: 40px;
  display: flex;
}

.clock {
  width: 25px;
  height: 25px;
}

.line {
  left: 237px;
  width: 8px;
  margin: 0 5px;
  border-top: 2px solid black;
}

.multilineText {
  position: absolute;
  left: 20px;
  width: 30px;
  height: 30px;
}

.close {
  width: 30px;
  height: 30px;
  cursor: pointer;
  margin-top: 10px;
}

.bascket {
  cursor: pointer;
}

.invalid {
  border: 1px solid red;
}