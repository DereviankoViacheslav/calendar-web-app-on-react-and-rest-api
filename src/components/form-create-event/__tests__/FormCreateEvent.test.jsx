import React from 'react';
import { shallow } from 'enzyme';
import FormCreateEvent from '../FormCreateEvent'

describe('<FormCreateEvent />', () => {
  const props = {
    onChange: jest.fn(),
    handleSubmit: jest.fn(),
    onClosePopup: jest.fn(),
    onDeleteEvent: jest.fn(),
    startDate: '2020-02-12',
    startTime: '12:00',
    endDate: '2020-02-12',
    endTime: '14:00',
    color: '#4183f1',
    name: 'Test event',
    description: 'Test description',
    isShowTrashIcon: true,
  };

  it('Get snapshot', () => {
    const wrappedComponent = shallow(<FormCreateEvent {...props} />);
    expect(wrappedComponent).toMatchSnapshot();
  });

  it('should render popup', () => {
    const wrappedComponent = shallow(<FormCreateEvent {...props} />);
    expect(wrappedComponent.find('.popup').exists()).toBeTruthy();
  });
});