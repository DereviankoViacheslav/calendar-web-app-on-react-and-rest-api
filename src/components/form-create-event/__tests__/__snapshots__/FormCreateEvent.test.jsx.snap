// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormCreateEvent /> Get snapshot 1`] = `
<form
  className="popup event"
  onSubmit={[MockFunction]}
>
  <span
    className="popup__btn-close"
    onClick={[MockFunction]}
  >
    <i
      className="fas fa-times"
    />
  </span>
  <input
    className="event__name"
    name="name"
    onChange={[MockFunction]}
    placeholder="Add title "
    required={true}
    type="text"
    value="Test event"
  />
  <div
    className="popup__color-picker"
  >
    <i
      className="fas fa-palette"
    />
    <label
      className="popup__color-picker_label"
    >
      <span
        className="popup__color-picker_label-text"
      >
        Select a color
      </span>
      <input
        className="event__color-picker"
        name="color"
        onChange={[MockFunction]}
        type="color"
        value="#4183f1"
      />
    </label>
  </div>
  <div
    className="popup__picker"
  >
    <span
      className="popup__picker-icon"
    >
      <i
        className="far fa-clock"
      />
    </span>
    <input
      className="event__date-start"
      name="startDate"
      onChange={[MockFunction]}
      required={true}
      type="date"
      value="2020-02-12"
    />
    <input
      className="event__time-start select"
      name="startTime"
      onChange={[MockFunction]}
      required={true}
      step="900"
      type="time"
      value="12:00"
    />
    <span
      className="decor-dash"
    />
    <input
      className="event__time-end select"
      name="endTime"
      onChange={[MockFunction]}
      required={true}
      step="900"
      type="time"
      value="14:00"
    />
    <input
      className="event__date-end"
      name="endDate"
      onChange={[MockFunction]}
      required={true}
      type="date"
      value="2020-02-12"
    />
  </div>
  <div
    className="popup__description"
  >
    <i
      className="fas fa-align-left"
    />
    <textarea
      className="event__description"
      name="description"
      onChange={[MockFunction]}
      placeholder="Add description"
      value="Test description"
    />
  </div>
  <div
    className="footer-popup"
  >
    <span
      className="event__btn-delete"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-trash-alt"
      />
    </span>
    <button
      className="event__btn-save"
    >
      Save
    </button>
  </div>
</form>
`;
